android-sdk/src/main/java/com/mobilecoin/lib/MobileCoinClient.java

```diff
@@ -221,11 +222,17 @@ public AccountSnapshot getAccountSnapshot(UnsignedLong blockIndex) throws Networ
         TxOutStore txOutStore = getTxOutStore();
         UnsignedLong storeIndex = txOutStore.getCurrentBlockIndex();
         if (storeIndex.compareTo(blockIndex) < 0) {
-            txOutStore.refresh(
-                    viewClient,
-                    ledgerClient,
-                    fogBlockClient
-            );
+            try {
+                txOutStore.refresh(
+                        viewClient,
+                        ledgerClient,
+                        fogBlockClient
+                );
+            } catch(FogSyncException e) {
+                if(blockIndex.compareTo(e.getFogBlockIndex()) >= 0) {
```

Comment:
Why block indexes check is needed after `FogSyncException` is already caught? Is it possible that it was thrown without a problem?
Also if this check is needed indeed, should it compare with `FOG_SYNC_THRESHOLD`?

Reply:
> Why block indexes check is needed after `FogSyncException` is already caught? Is it possible that it was thrown without a problem? Also if this check is needed indeed, should it compare with `FOG_SYNC_THRESHOLD`?

Let's say consensus is at block 30, fog is at block 20, and I want a snapshot of block 10. When refresh is called, it will throw the sync exception because fog is off by the threshold. But the txoutstore was still updated to block 20. Therefore I can say that a snapshot at block 10 is valid. This is probably a rare scenario, but I don't consider it a problem in this case. That is what this check is for.